File: Design_Pattern_and_Principles/AdapterPatternExample/AdapterTest.java
public class AdapterTest {
    public static void main(String[] args) {
        PaymentProcessor processor1 = new PayPalAdapter(new PayPalGateway());
        PaymentProcessor processor2 = new StripeAdapter(new StripeGateway());

        System.out.println("--- Testing Adapters ---");
        processor1.processPayment(49.99);
        processor2.processPayment(10.00);
    }
}

File: Week_1/Design_Pattern_and_Principles/AdapterPatternExample/AdapterTest.cs
using System;

public class AdapterTest {
    public static void Main(string[] args) {
        IPaymentProcessor processor1 = new PayPalAdapter(new PayPalGateway());
        IPaymentProcessor processor2 = new StripeAdapter(new StripeGateway());

        Console.WriteLine("--- Testing Adapters ---");
        processor1.ProcessPayment(49.99);
        processor2.ProcessPayment(10.00);
    }
}